ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -190,26 +205,38 @@ public struct SignalProducer<T, E: ErrorType> {
 		// Create a composite disposable that will automatically be torn
 		// down when the signal terminates.
 		let compositeDisposable = CompositeDisposable()
-		compositeDisposable.addDisposable {
-			// Upon early disposable, interrupt the observer and all dependents.
-			sendInterrupted(observer)
```

Comment:
This particular method is an excellent illustration of how confusing recursive sending can be.

Before this PR, disposal could send you an `.Interrupted` event _while you’re processing another_. I seriously doubt any extant RAC code handles this case except by accident.

With the changes here, disposal waits until you’re done with your event processing.